torch/_guards.py

```diff
@@ -536,7 +536,11 @@ def restore_graphstate(self, state):
 
 class CompileContext:
     @staticmethod
-    def get() -> Optional[CompileContext]:
+    def get() -> CompileContext:
+        return _TLS.compile_context
```

Comment:
You need to explicitly assert that compile_context is non-None here; by default we reset compile_context to None when we're done, rather than del it.